From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 12 Dec 2024 19:13:16 -0600
Subject: [PATCH] solus: Set default settings for compression and build-id

Set the default to compress debug sections with zstd, and also set the default to create a build-id using BLAKE3 that's 160 bits long.
---
 src/mold.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/mold.h b/src/mold.h
index ce74a26e..4de962f4 100644
--- a/src/mold.h
+++ b/src/mold.h
@@ -1771,9 +1771,9 @@ struct BuildId {
     }
   }
 
-  enum { NONE, HEX, HASH, UUID } kind = NONE;
+  enum { NONE, HEX, HASH, UUID } kind = HASH;
   std::vector<u8> value;
-  i64 hash_size = 0;
+  i64 hash_size = 20;
 };
 
 typedef enum { COMPRESS_NONE, COMPRESS_ZLIB, COMPRESS_ZSTD } CompressKind;
@@ -1895,7 +1895,7 @@ struct Context {
     BsymbolicKind Bsymbolic = BSYMBOLIC_NONE;
     BuildId build_id;
     CetReportKind z_cet_report = CET_REPORT_NONE;
-    CompressKind compress_debug_sections = COMPRESS_NONE;
+    CompressKind compress_debug_sections = COMPRESS_ZSTD;
     MultiGlob undefined_glob;
     SeparateCodeKind z_separate_code = NOSEPARATE_CODE;
     ShuffleSectionsKind shuffle_sections = SHUFFLE_SECTIONS_NONE;
